pseudomonas_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -304,6 +311,132 @@
+
+
+# In[24]:
+
+
+print(template_enriched_pathways.shape)
+template_enriched_pathways[template_enriched_pathways['padj'] < 0.05].sort_values(by='padj')
+
+
+# In[25]:
+
+
+# Create "<local_dir>/GSEA_stats/" subdirectory
+os.makedirs(os.path.join(local_dir, "GSEA_stats"), exist_ok=True)
+
+
+# In[26]:
+
+
+get_ipython().run_cell_magic('R', '-i project_id -i local_dir -i adage_kegg_DB_processed_filename -i num_runs -i statistic', '\nsource(paste0(base_dir, \'/generic_expression_patterns_modules/GSEA_analysis.R\'))\n\n# New files created: "<local_dir>/GSEA_stats/GSEA_stats_simulated_data_<project_id>_<n>.txt"\nfor (i in 0:(num_runs-1)) {\n    simulated_DE_stats_filename <- paste(local_dir, \n                                     "DE_stats/DE_stats_simulated_data_", \n                                     project_id,\n                                     "_", \n                                     i,\n                                     ".txt",\n                                     sep = "")\n    \n    out_filename <- paste(local_dir, \n                     "GSEA_stats/GSEA_stats_simulated_data_",\n                     project_id,\n                     "_",\n                     i,\n                     ".txt", \n                     sep = "")\n    \n    enriched_pathways <- find_enriched_pathways(simulated_DE_stats_filename, adage_kegg_DB_processed_filename, statistic) \n    \n    # Remove column with leading edge since its causing parsing issues\n    write.table(as.data.frame(enriched_pathways[1:7]), file = out_filename, row.names = F, sep = "\\t")\n}')
+
+
 # ### Rank pathways 
 
+# In[27]:
+
+
+# Concatenate simulated experiments
+simulated_GSEA_stats_all = process.concat_simulated_data(local_dir, num_runs, project_id, 'GSEA')
+simulated_GSEA_stats_all.set_index('pathway', inplace=True)
+print(simulated_GSEA_stats_all.shape)
+
+
+# In[28]:
+
+
+# Aggregate statistics across all simulated experiments
+simulated_GSEA_summary_stats = calc.aggregate_stats(
+    col_to_rank_pathways,
+    simulated_GSEA_stats_all,
+    'GSEA'
+)
+
+simulated_GSEA_summary_stats.sort_values(by=('padj', 'median')).head()
+
+
+# In[29]:
+
+
+# Load association statistics for template experiment
+template_GSEA_stats = template_enriched_pathways.iloc[:, :-1]
+template_GSEA_stats.set_index('pathway', inplace=True)
+
+template_GSEA_stats.head()
+
+# Rank genes in template experiment
+template_GSEA_stats = calc.rank_genes_or_pathways(
+    col_to_rank_pathways,
+    template_GSEA_stats,
+    True
+)
+
+
+# In[30]:
+
+
+# Rank genes in simulated experiments
+simulated_GSEA_summary_stats = calc.rank_genes_or_pathways(
+    col_to_rank_pathways,
+    simulated_GSEA_summary_stats,
+    False
+)
+
+
 # ### Pathway summary table
+
+# In[31]:
+
+
+# Create intermediate file: "<local_dir>/gene_summary_table_<col_to_rank_pathways>.tsv"
+summary_pathway_ranks = process.generate_summary_table(
```

Comment:
Would it make sense to do a pair plot here of the rank in the template experiment and the average rank in the simulated experiments (a bit like the one you did for the human data, except I think you were comparing against the processed data from Powers et al)?

Or are you not interested in how similar the template and simulated ranks are for this experiment?

Reply:
For the pseudomonas dataset, we don't have a reference to compare against like with the human dataset. The difference between the template vs simulated ranks is summarized in the z-score, which says how much the template test statistic deviates from the simulated test statistic.

Reply:
Ohhh, got it. Do you have any other ideas of ways to validate these results? Or has nobody studied generic gene expression in Pseudomonas (and you're trying to discover new things here)?

Reply:
So the idea here was that we validated that our method can identify generic genes using the human datasets, and this pseudomonas analysis is more for discovery of specific and generic gene candidates that we want to test in the lab thanks to the Hogan lab